crates/cheatcodes/src/inspector.rs

```diff
@@ -1514,6 +1538,225 @@ impl<DB: DatabaseExt> Inspector<DB> for Cheatcodes {
 
         outcome
     }
+
+    fn eofcreate(
+        &mut self,
+        ecx: &mut EvmContext<DB>,
+        call: &mut EOFCreateInput,
+    ) -> Option<EOFCreateOutcome> {
+        let ecx = &mut ecx.inner;
+        let gas = Gas::new(call.gas_limit);
+
```

Comment:
can we pls minimize code dedup between these new methods and legacy? This entire file is already complex enough

Reply:
@DaniPopes make sense, done here https://github.com/foundry-rs/foundry/pull/8123/files/46be51fca66102f9cc41d079e9c2ec2886e42bde..b0e9714fd9b5b77c5ff26a3883e6b4a5b1d899f1 ptal